WS2012R2/lisa/setupscripts/TCUtils.ps1

```diff
@@ -1311,11 +1311,62 @@ function check_app([string]$app_name, [string]$custom_ip)
     }
     .\bin\plink -i ssh\${sshKey} root@${target_ip} "command -v ${app_name}"
     if (-not $?) {
-        Write-Output "ERROR: ${app_name} is not installed on the VM" -ErrorAction SilentlyContinue | Out-File -Append $summaryLog
         return $False
     }
+    return $True
+}
+
+#######################################################################
+#
+# install application on VM
+#
+#######################################################################
+
+function install_app([string]$app_name, [string]$custom_ip, [string]$app_githubLink, [string]$app_version)
+{
+    if ($app_name -eq $null -and $app_githubLink -eq $null)
+    {
+        Write-Output "ERROR: ${app_name} or $app_githubLink is not set" -ErrorAction SilentlyContinue | Out-File -Append $summaryLog
+        return $False
+    }
+
+    If([string]::IsNullOrWhiteSpace($custom_ip)) {
+        $target_ip = $ipv4
+    }
+    else {
+        $target_ip = $custom_ip
+    }
+    # check whether app is already installed
+    $retVal = check_app $app_name $target_ip
+    if ($retVal)
+    {
+        return $True
+    }
+
+    # app is not insalled, install it
+    if ($app_name -eq "stress-ng")
+    {
+        $app_githubLink="https://github.com/ColinIanKing/stress-ng"
+        $app_version="V0.07.16"
+    }
+    elseif ($app_name -eq "stressapptest")
+    {
+        $app_githubLink="https://github.com/stressapptest/stressapptest.git"
+    }
+
+    .\bin\plink -i ssh\${sshKey} root@${target_ip} "cd /root; git clone $app_githubLink $app_name"
+
+    if ($app_version)
```

Comment:
would it be possible to script the installation of the tools independently, similar to the aap_name if loop?
This was we can re-use the install_app function later on

Reply:
@chvalean thank you for your comment.

I think I can script the installation of the tools, but need some guidance about how to implement it, I'm not sure the requirement. Could you please help to explain more?

1) "For independently", do you mean "install_app" function itself, I need to remove check_app or put it in other place? 
2) I'm not very clear about "similar to the app_name if loop", you mean support app name as list, then can install multiple app at the same time, such as put the app_name and github link, github version to dictionary, then the list includes dictionary?  Currently if app_name is not stress-ng or stressapptest, it also can support, but when provide app name, needs to provide git link.

Thank you so much.

Reply:
I was thinking at having install_app in a more universal way, so that in the future we can extend it to install other tools as well.
Currently with the new changes, it is more or less hardcoded to handle only stress-ng and stressapptest with their specific method of installation.

One suggestion would be to encapsulate any tool installation in the name parsed - eg if ($app_name -eq "stress-ng")